tests/test_morphing.py

```diff
@@ -0,0 +1,222 @@
+
+    Nd = 0
+    Np = 0
+    Ns = 1
+    assert new_morpher.find_components(Ns = Ns, Nd = Nd, Np= Np).any() == np.array([[2,2]]).any(), " The output differs from expected components"   
+
+    Nd = 2
+    Np = 1
+    Ns = 0
+    assert new_morpher.find_components(Ns = Ns, Nd = Nd, Np= Np).any()== np.array([[2,0,2], [1,1,2], [0,2,2]]).any(), " The output differs from expected components"   
+
+
+def weight_gradients(gs = np.array([[1,1,1,1,1], [-5, -4, -3, -2, -1]]),
+                    gp = None, gd = None, parameter_max_power = [2,2],
+                    this_components = np.array([[4, 0], [3, 1], [2, 2], [1, 3], [0, 4]]),
+                    this_basis = np.array([[1, -5], [1, -4], [1, -3], [1, -2], [1, -1]])):
+
+    # Morpher object with the new format of inputs
+    new_morpher = m.PhysicsMorpher(parameter_max_power=parameter_max_power, parameter_range=[(-1.,1.,), (-1.,1.)])
+
+    _ = new_morpher.find_components(Ns = 2, Nd = 0, Np= 0)
+    new_morpher.set_basis(basis_d = gd, basis_p = gp, basis_s=gs)
+
+    # Morpher object with the initial format of inputs
+    old_morpher = m.PhysicsMorpher(parameter_max_power=[2,2])
+    old_morpher.set_components(this_components)
+    old_morpher.set_basis(basis_numpy= this_basis)
+
+    # calculate and check the output values of the old and new mophing weights. 
+    old_weight_gradient = old_morpher.calculate_morphing_weight_gradient(theta = [1,1])
+    new_weight_gradient = new_morpher.calculate_morphing_weight_gradient(theta = [1,1])
+
+    assert old_weight_gradient.any() == new_weight_gradient .any(), "The weight gradient of old inputs and the new inputs are different."
+
+
+def evaluate_morphing():
+    # Check if the morpher object created with old and new inputs can execute evaluate_morphing() function
+    old_morpher = m.PhysicsMorpher(parameter_max_power=[4,4],
+                    parameter_range=[(-1.,1.), (-1.,1.)])
+    this_components = np.array([[4, 0], [3, 1], [2, 2], [1, 3], [0, 4]]) 
+    this_basis = np.array([[1, -5], [1, -4], [1, -3], [1, -2], [1, -1]])
+
+    # Set the powers of components and basis
+    old_morpher.set_components(this_components)
+    old_morpher.set_basis(basis_numpy= this_basis)
+
+    old_value = old_morpher.evaluate_morphing(basis = this_basis)
+
+    # Morpher object with the new format of inputs
+    new_morpher = m.PhysicsMorpher(parameter_max_power=[4,4], parameter_range=[(-1.,1.,), (-1.,1.)])
+
+    gd = None 
+    gp = None 
+    gs = np.array([[1,1,1,1,1], [-5, -4, -3, -2, -1]])
+
+    _ = new_morpher.find_components(Ns = 2, Nd = 0, Np= 0)
+    new_morpher.set_basis(basis_d = gd, basis_p = gp, basis_s=gs)
+
+    new_value = new_morpher.evaluate_morphing(basis = this_basis)
+
+    assert np.abs(new_value - old_value)<10**3, "The output values of old and new evaluate_morphing with same inputs varies more than 10^3"
```

Comment:
[`np.allclose`](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) is a convenient way to do this kind of check.

Where does the randomness come from that causes the results to change?

Reply:
I think evaluate_morphing will randomly draw n_test_thetas within the given range. The output also could differs greater than 100 in different run of the original code. 
It was on 819 that called the helper method _draw_random_thetas(n_thetas=n_test_thetas), and in helper method on line 878 it will call np.random.rand(n_thetas, self.n_parameters)

Reply:
I see that this is part is no longer included, so would that instead go in via #506 or is that functionality covered elsewhere?

Reply:
I will add it via #506, I think it won't be able to run if morphing is not changed